pimd/pim6_cmd.c

```diff
@@ -104,6 +104,77 @@ DEFUN (show_ipv6_mld_groups_vrf_all,
     return CMD_SUCCESS;
 }
 
+DEFUN (show_ipv6_mld_interface,
+       show_ipv6_mld_interface_cmd,
+       "show ipv6 mld [vrf NAME] interface [detail|WORD] [json]",
+       SHOW_STR
+       IPV6_STR
+       MLD_STR
+       VRF_CMD_HELP_STR
+       "MLD interface information\n"
+       "Detailed output\n"
+       "interface name\n"
+       JSON_STR)
+{
+    int idx = 2;
+    struct vrf *vrf = pim_cmd_lookup_vrf(vty, argv, argc, &idx);
+    bool uj = use_json(argc, argv);
+
+    if (!vrf)
+        return CMD_WARNING;
+
+    if (argv_find(argv, argc, "detail", &idx)
+        || argv_find(argv, argc, "WORD", &idx))
+        //To be done
+        //mld_show_interfaces_single()
+    else
+        //To be done
+        //mld_show_interfaces()
+
+    return CMD_SUCCESS;
+}
+
+DEFUN (show_ipv6_mld_interface_vrf_all,
+       show_ipv6_mld_interface_vrf_all_cmd,
+       "show ipv6 mld vrf all interface [detail|WORD] [json]",
+       SHOW_STR
+       IPV6_STR
+       MLD_STR
+       VRF_CMD_HELP_STR
+       "MLD interface information\n"
+       "Detailed output\n"
+       "interface name\n"
+       JSON_STR)
+{
+    int idx = 2;
+    bool uj = use_json(argc, argv);
+    struct vrf *vrf;
+    bool first = true;
+
+    if (uj)
+        vty_out(vty, "{ ");
+    RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name) {
+        if (uj) {
```

Comment:
This is wrong. Seems base code is also wrong but lets not do it this way. We need to add vrf as an object. refer ospfv3 show commands. Sai knows this.